client/components/UserSettings/ConfigureUserSettingsModal.tsx

```diff
@@ -66,6 +74,19 @@ export const ConfigureUserSettingsModal: React.FC<ModalProps> = ({ show, onClose
         });
     }, [updateSettings, country, interestedInSources, enableEmailNotifications, email]);
 
+    const { withRequest, handleError } = useHttp();
+
+    const deleteUser = useCallback(async () => {
+        await withRequest(async http => {
+            await (await http.delete(`/user`));
+        });
+    }, [withRequest]);
+
+    const cancelAlertRef = React.useRef(null);
+    const [showDialog, setShowDialog] = React.useState(false);
+    const closeAlert = () => setShowDialog(false);
+    const openAlert = () => setShowDialog(true);
```

Comment:
Instead of these manual wrappers you could use a hook called `useDisclosure` that will give you the helpers and the state. See  https://github.com/Agreon/game-watch/blob/c85c38d07b7d99aa138dd3c9e8cd496bf4e28625/client/components/Header.tsx#L38-L42 for reference.

Reply:
I did originally want to do that, but it marked onClose as a duplicate of line 38's onClose. Doing it in the similar fashion as the reference you posted will solve that problem though I believe.